1-callbacks/solution.js

```diff
@@ -31,17 +31,54 @@ node solution.js name1 name2 name3
 5. another challenge is: after you solve the challenge using callback style, in another file promisify the callback and solve it again
 ** give a look to node.js util.promisify, avoid to alter the validate-user.file **
 */
+const validateUser = require("./validate-user.js");
 
 function solution() {
-    // YOUR SOLUTION GOES HERE
-
     // you get your 5 names here
+    const names = ["Carlos", "Andrea", "Miguel", "Juan", "Richard"];
+
+    // extra challange #1 read from terminal arguments
+    if(process.argv.length > 2){
+        // just add them, so that I don't have to type a bunch in terminal. 
+        names.push(...process.argv.slice(2));
+    }
+    
+    const okNames = [];
+    const errNames = [];
+    let cbCount = 0 ; // console results are shown before all cbs finish, so I will add logic to avoid showing until last cb
+
+    // YOUR SOLUTION GOES HERE
+    const myCallback = (...params)=>{ // expand because if is in list it sends 2
```

Comment:
The callback name could be improve

Reply:
You can also define the callback function outside of this function to keep your implementation cleaner.